File: abc/abc122/abc122_c.cs
// https://atcoder.jp/contests/abc122/submissions/9267194

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var input = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
            var N = input[0];
            var Q = input[1];

            var S = Console.ReadLine();

            var lr = new int[Q][];
            for (int i = 0; i < Q; i++)
            {
                lr[i] = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
            }

            var t = new int[N + 1];
            for (int i = 1; i < N; i++)
            {
                if (S[i - 1] == 'A' && S[i] == 'C')
                {
                    t[i + 1] = t[i] + 1;
                }
                else
                {
                    t[i + 1] = t[i];
                }
            }

            for (int i = 0; i < Q; i++)
            {
                Console.WriteLine(t[lr[i][1]] - t[lr[i][0]]);
            }
        }
    }
}

File: abc/abc122/abc122_c.fs
// https://atcoder.jp/contests/abc122/submissions/12205358

let [| N; Q |] = stdin.ReadLine().Split() |> Array.map int
let S = stdin.ReadLine()

let lr = Array.zeroCreate<int []> Q

for i in [ 0..Q - 1 ] do
    lr.[i] <- stdin.ReadLine().Split() |> Array.map int

let t = Array.zeroCreate<int> (N + 1)

for i in [ 1..N - 1 ] do
    if S.[i - 1] = 'A' && S.[i] = 'C' then t.[i + 1] <- t.[i] + 1
    else t.[i + 1] <- t.[i]

for i in [ 0..Q - 1 ] do
    t.[lr.[i].[1]] - t.[lr.[i].[0]] |> stdout.WriteLine
